com.gamefuse.sdk/Runtime/GameFuse.cs

```diff
@@ -362,10 +362,11 @@ private IEnumerator SignUpRoutine(string email, string password, string password
                 GameFuseUser.CurrentUser.SetCreditsInternal(Convert.ToInt32(json.GetNumber("credits")));
                 GameFuseUser.CurrentUser.SetUsernameInternal(json.GetString("username"));
                 GameFuseUser.CurrentUser.SetLastLoginInternal(DateTime.Parse(json.GetString("last_login")));
-                GameFuseUser.CurrentUser.SetNumberOfLoginsInternal(Convert.ToInt32(json.GetNumber("number_of_logins")));
+                GameFuseUser.CurrentUser.SetNumberOfLoginsInternal(Convert.ToInt32(json.GetNumber("number_of_logins"))); 
                 GameFuseUser.CurrentUser.SetAuthenticationTokenInternal(json.GetString("authentication_token"));
                 GameFuseUser.CurrentUser.SetIDInternal(Convert.ToInt32(json.GetNumber("id")));
-                GameFuseUser.CurrentUser.DownloadAttributes(true, callback); // Chain next request - download users attributes
+                GameFuseUser.CurrentUser.DownloadAttributes(true, callback); // Chain next request - download users attributes  
```

Comment:
In V3 api, we no longer need to chain the download attributes sequence.  Please test the login and correct this.  Login will have the fields included in "@show_full_data" for logged in user.  Other users downloaded will not have this as we download just their info not their full data. You can hit the show route on another user to gather the rest of their data.  Here is the view from the server.  You can see we no longer need to chain requests:

```
# frozen_string_literal: true

json.partial! 'api/v3/users/user', user: @user_to_show # just the basic attributes of id, username, email, display_email, credits, score
if @for_current_user
  json.last_login @user_to_show.last_login || Time.now
  json.number_of_logins @user_to_show.number_of_logins || 0
  json.authentication_token @user_to_show.authentication_token
end

if @show_full_data

  #move this to only for_current_user
  json.game_sessions_current_month @user_to_show.game_sessions.where(created_at: Time.now.at_beginning_of_month..Time.now).size
  json.game_sessions_total @user_to_show.game_sessions.size
  json.events_total @user_to_show.user_events.size
  json.events_current_month @user_to_show.user_events.where(created_at: Time.now.at_beginning_of_month..Time.now).size

  json.game_user_attributes @user_to_show.game_user_attributes do |game_user_attribute|
    json.key game_user_attribute.key
    json.value game_user_attribute.decrypt_value
  end

  json.game_user_store_items @user_to_show.game_user_store_items do |game_user_store_item|
    store_item = game_user_store_item.store_item
    json.extract!(store_item, :name, :category, :description, :icon_url, :cost, :id)
  end

  json.leaderboard_entries @user_to_show.leaderboard_entries do |leaderboard_entry|
    json.extract!(leaderboard_entry, :score, :leaderboard_name, :extra_attributes, :created_at)
  end

  # Split this, only show /friends here, move incoming and outgoing requests to only signed in section & document this
  json.partial! 'api/v3/friendships/friendship_data' # the data inside of this will only render if key variables are present

  json.groups @groups do |group|
    json.partial! 'api/v3/groups/group', group: group, with_full_data: false
  end

  json.game_rounds @game_rounds do |game_round|
    json.partial! 'api/v3/game_rounds/game_round', game_round: game_round
  end

  if @for_current_user
    
    json.group_join_requests @join_requests do |join_request|
      json.extract!(join_request, :id, :status)
      json.user { json.partial! 'api/v3/users/user', user: @user_to_show }
      json.group { json.partial! 'api/v3/groups/group', group: join_request.group, with_full_data: false }
    end

    json.group_invites @group_invites do |invite|
      json.extract!(invite, :id, :status)
      json.user { json.partial! 'api/v3/users/user', user: @user_to_show }
      json.inviter { json.partial! 'api/v3/users/user', user: invite.inviter }
      json.group { json.partial! 'api/v3/groups/group', group: invite.group, with_full_data: false }
    end

    json.direct_chats @direct_chats do |chat|
      json.partial! 'api/v3/chats/chat', chat: chat, chat_users: chat.game_users
    end

    json.group_chats @group_chats do |chat|
      json.partial! 'api/v3/chats/chat', chat: chat, chat_users: chat.group.members
    end
  end
end
```

Reply:
Let me know if you have thoughts on this pattern. I am open to changing it on the server side. Let me know what you think is best?

Reply:
I captured this a a card on trello.